dask/array/ufunc.py

```diff
@@ -223,6 +229,8 @@ def outer(self, A, B, **kwargs):
 less_equal = ufunc(np.less_equal)
 not_equal = ufunc(np.not_equal)
 equal = ufunc(np.equal)
+isneginf = partial(equal, -np.inf)
+isposinf = partial(equal, np.inf)
```

Comment:
Can you please remind me why these are needed?

Reply:
Sure, these are not ufuncs:

```python
In [1]: import numpy as np

In [2]: isinstance(np.equal, np.ufunc)
Out[2]: True

In [3]: isinstance(np.isposinf, np.ufunc)
Out[3]: False
```

Reply:
Right. Am more curious why these have been added to replace their `wrap_elemwise` equivalents below. Just to clarify am not objecting to doing this if it helps move things along. Merely interested in why it was needed is all.

Reply:
So when `__array_function__` is enabled, those are dispatched through that. Libraries such as sparse don't implement those directly, so it gives us better compatibility in such cases.

Reply:
Sparse doesn't implement `__array_function__` or doesn't implement `isneginf` and `isposinf`?

Reply:
Sparse doesn't implement `isneginf` and `isposinf`.

Reply:
Thanks for the info. Have raised issue ( https://github.com/pydata/sparse/issues/252 ) on that point. Happy to carry this workaround for now.

Reply:
I don't know if we want to have that implemented in Sparse. I was only using Sparse as an example, CuPy also doesn't implement it.

There's another good point (at least on the NumPy side), it is faster compute that using `np.equal`, as shown [here](https://github.com/numpy/numpy/issues/13100#issuecomment-470310126).

 In my opinion, it should remain as it is in this PR, seems faster and more reliable when dealing with other NumPy-like libraries.

Reply:
Well I think one outcome of the `__array_function__` protocol is it will put pressure on other libraries to implement a fully compatible NumPy API for their arrays. Currently for us today that is seeing `isneginf` and `isposinf` implemented in Sparse and CuPy. Maybe it will be something else later.

That to me is a sign that NumPy should improve their `isneginf` and `isposinf` implementations. Not that the API should be dispensed with completely. I don't think we can maintainably tweak our implementations based on this sort of thing because some function we special case could be suboptimal for some other library (either for performance or something totally different).

As noted above am not objecting to having this here for now. Have also already approved the PR. Though I think long term we shouldn't rely on this sort of thing. Perhaps in this case we put some pressure on NumPy to implement `isneginf` and `isposinf` in a more friendly way (as Marten showed in that comment).

Reply:
Actually I forgot [Eric's earlier comment in that thread]( https://github.com/numpy/numpy/issues/13100#issuecomment-470272845 ). We might be able to just alias NumPy's implementations as it already leverages the `__array_ufunc__` protocol under the hood. Anyways this is work for another day (not this PR). 🙂

Reply:
You make a fair point @jakirkham. I was simply putting that given the current conditions, we may be better off with the new implementation (from a performance point of view), but you're probably right, the long-term solution is indeed to improve the NumPy implementations.